support-frontend/assets/pages/[countryGroupId]/components/formOnSubmit.ts

```diff
@@ -31,6 +31,16 @@ import { createSubscription } from './createSubscription';
 import type { PaymentMethod } from './paymentFields';
 import { FormSubmissionError } from './paymentFields';
 
+const getConsentValue = (formData: FormData, userWasShownCheckbox: boolean) => {
+	const similarProductsCheckbox = formData.get('similarProductsConsent');
+	if (similarProductsCheckbox) {
+		return similarProductsCheckbox === 'on';
+	}
```

Comment:
i guess this will always be true if `similarProductsCheckbox`is `on` because unchecked never shows up.
we could 
```suggestion
	    const similarProductsCheckbox = formData.get('similarProductsConsent');
      return similarProductsCheckbox === 'on';
	}
```